src/main/scala/com/rbmhtechnology/eventuate/sandbox/ReplicationEndpoint.scala

```diff
@@ -0,0 +1,167 @@
+package com.rbmhtechnology.eventuate.sandbox
+
+import java.util.concurrent.atomic.AtomicReference
+import java.util.function.UnaryOperator
+
+import akka.actor._
+import akka.pattern.{ask, pipe}
+
+import com.rbmhtechnology.eventuate.sandbox.ReplicationFilter.NoFilter
+import com.rbmhtechnology.eventuate.sandbox.ReplicationProtocol._
+import com.typesafe.config._
+
+import scala.collection.immutable.Seq
+import scala.concurrent.Future
+
+object ReplicationEndpoint {
+  def logId(endpointId: String, logName: String): String =
+    s"${endpointId}_$logName"
+}
+
+class ReplicationEndpoint(
+  val id: String,
+  logNames: Set[String],
+  targetFilters: Map[String, ReplicationFilter],
+  sourceFilters: Map[String, ReplicationFilter],
+  config: Config = ConfigFactory.empty()) {
+
+  import ReplicationEndpoint._
+
+  private val _connections: AtomicReference[Map[String, Set[ActorRef]]] =
+    new AtomicReference(Map.empty)
+
+  val system: ActorSystem =
+    ActorSystem(s"$id-system", config)
+
+  val settings: ReplicationSettings =
+    new ReplicationSettings(system.settings.config)
+
+  val eventLogs: Map[String, ActorRef] =
+    logNames.foldLeft(Map.empty[String, ActorRef]) {
+      case (logs, logName) => logs + (logName -> createEventLog(logName))
+    }
+
+  val connectionAcceptor: ActorRef =
+    createConnectionAcceptor()
+
+  import system.dispatcher
+
+  def connections: Set[String] =
+    _connections.get.keySet
+
+  def connect(remoteEndpoint: ReplicationEndpoint): Future[String] =
+    connect(remoteEndpoint.connectionAcceptor)
+
+  def connect(remoteAcceptor: ActorRef): Future[String] =
+    remoteAcceptor.ask(GetReplicationSourceLogs(logNames))(settings.askTimeout).mapTo[GetReplicationSourceLogsSuccess].map { reply =>
+      val replicators = reply.sourceLogs.map {
+        case (logName, sourceLog) =>
+          val sourceLogId = logId(reply.endpointId, logName)
+          val targetLogId = logId(id, logName)
+          createReplicator(sourceLogId, sourceLog, targetLogId, eventLogs(logName))
+      }
+      addConnection(reply.endpointId, replicators.toSet)
+      reply.endpointId
+    }
+
+  def disconnect(remoteEndpointId: String): Unit =
+    removeConnection(remoteEndpointId).foreach(system.stop)
+
+  def terminate(): Future[Terminated] =
+    system.terminate()
```

Comment:
Individual replicators can be stopped with `disconnect`. The endpoint-specific actor system must be terminated with the endpoint itself as its creation is also bound to endpoint creation.